xtext-gradle-plugin/src/integTest/java/org/xtext/gradle/test/BuildingANonStandardXtendProject.xtend

```diff
@@ -0,0 +1,80 @@
+package org.xtext.gradle.test
+
+import org.junit.Ignore
+import org.junit.Test
+
+import static org.hamcrest.core.IsEqual.equalTo
+import static org.junit.Assert.*
+import static org.junit.Assume.*
+
+/**
+ * Tests for Xtend projects that do not use the default configuration.
+ */
+class BuildingANonStandardXtendProject extends AbstractIntegrationTest {
+
+	override setup() {
+		super.setup()
+		buildFile << xtendPluginSnippet
+	}
+
+	@Test
+	def void compilesToJava8WhenConfigured() {
+		assumeTrue(System.getProperty('java.version').startsWith('1.8'))
+
+		// given 
+		createXtendHelloWorld
+		buildFile << '''
+			xtext.languages.xtend.generator.javaSourceLevel = "1.8"
+		'''
+
+		// when
+		build('generateXtext')
+
+		// then
+		val generatedJava = file('build/xtend/main/HelloWorld.java').contentAsString
+		assertTrue(generatedJava.contains('import java.util.function.Consumer;'))
+	}
+
+	@Test @Ignore
+	def void compilesToJava7WhenConfigured() {
+		// given 
+		createXtendHelloWorld
+		buildFile << '''
+			xtext.languages.xtend.generator.javaSourceLevel = "1.7"
+		'''
+
+		// when
+		build('generateXtext')
+
+		// then
+		val generatedJava = file('build/xtend/main/HelloWorld.java').contentAsString
+		assertFalse(generatedJava.contains('import java.util.function.Consumer;'))
```

Comment:
This is not testing what you want to test =)

SourceLevel and API are two different things. The SourceLevel controls which Java features we use. So when you compile a Lambda in Java 7, it should become an anonymous class. When you compile it in Java 8 it should become a Java Lambda.

No matter which SourceLevel you choose, if you have JDK 8 on the classpath, we will link against Iterable.foreEach instead of IterableExtensions.forEach, because members take precedence over extensions. Unfortunately, the JDK classes have no information in them that tells us "this was added in version X".